tests/mocking/io.py

```diff
@@ -117,11 +119,15 @@ class WriteKeyedKafkaIO(UnifiedIO):
 
 
 class MockKafkaProducer:
-    def __init__(self):
+    def __init__(self, **kwargs):
         self.my_stream = []
+        self.config = kwargs  # Store the config to ensure it can accept any options
+        self.produce_call_count = 0
 
-    def send(self, topic: str, value: dict, key: str = None):  # pylint: disable=unused-argument
+    def produce(self, topic: str, key: Optional[bytes], value: bytes, on_delivery: Callable):  # pylint: disable=unused-argument
```

Comment:
Is there a reason you're mocking out the kafka calls? Can we not have an actual kafka deployed alongside in CI?

Reply:
Great idea... though needs more effort. happy to implement in a later PR!